spec/controllers/report_controller_spec.rb

```diff
@@ -2,128 +2,180 @@
       end
     end
 
-    context 'mark post report as admin' do
+    context "mark post report as admin" do
       before do
         Role.add_admin(alice.person)
       end
-      it 'succeeds' do
-        put :update, :id => @message.id, :type => 'post'
+      it "succeeds" do
+        put :update, id: @message.id, type: "post"
         expect(response.status).to eq(302)
-        expect(Report.where(:reviewed => true, :item_id => @message.id, :item_type => 'post')).to be_truthy
+        expect(Report.where(reviewed: true, item_id: @message.id, item_type: "post")).to be_truthy
       end
     end
-    context 'mark comment report as admin' do
+    context "mark comment report as admin" do
       before do
         Role.add_admin(alice.person)
       end
-      it 'succeeds' do
-        put :update, :id => @comment.id, :type => 'comment'
+      it "succeeds" do
+        put :update, id: @comment.id, type: "comment"
         expect(response.status).to eq(302)
-        expect(Report.where(:reviewed => true, :item_id => @comment.id, :item_type => 'comment')).to be_truthy
+        expect(Report.where(reviewed: true, item_id: @comment.id, item_type: "comment")).to be_truthy
+      end
+    end
+
+    context "mark post report as moderator" do
+      before do
+        Role.add_moderator(alice.person)
+      end
+      it "succeeds" do
+        put :update, id: @message.id, type: "post"
+        expect(response.status).to eq(302)
+        expect(Report.where(reviewed: true, item_id: @message.id, item_type: "post")).to be_truthy
+      end
+    end
+    context "mark comment report as moderator" do
+      before do
+        Role.add_moderator(alice.person)
+      end
+      it "succeeds" do
+        put :update, id: @comment.id, type: "comment"
+        expect(response.status).to eq(302)
+        expect(Report.where(reviewed: true, item_id: @comment.id, item_type: "comment")).to be_truthy
       end
     end
   end
 
-  describe '#destroy' do
-    context 'destroy post as user' do
-      it 'is behind redirect_unless_admin' do
-        delete :destroy, :id => @message.id, :type => 'post'
+  describe "#destroy" do
+    context "destroy post as user" do
+      it "is behind redirect_unless_admin_or_moderator" do
+        delete :destroy, id: @message.id, type: "post"
         expect(response).to redirect_to stream_path
-        expect(Report.where(:reviewed => false, :item_id => @message.id, :item_type => 'post')).to be_truthy
+        expect(Report.where(reviewed: false, item_id: @message.id, item_type: "post")).to be_truthy
       end
     end
-    context 'destroy comment as user' do
-      it 'is behind redirect_unless_admin' do
-        delete :destroy, :id => @comment.id, :type => 'comment'
+    context "destroy comment as user" do
+      it "is behind redirect_unless_admin_or_moderator" do
+        delete :destroy, id: @comment.id, type: "comment"
```

Comment:
I'd rather use lazy loaded "lets" instead of boring old instance variables here and everywhere else: http://betterspecs.org/#let